File: ios/Zest/Services/RecipeImportService.swift
import Foundation

class RecipeImportService {
    func importFromUrl(_ urlString: String) async throws -> [String: Any] {
        guard let url = URL(string: urlString) else {
            throw NSError(domain: "Invalid URL", code: 0)
        }
        
        let (data, _) = try await URLSession.shared.data(from: url)
        let html = String(data: data, encoding: .utf8) ?? ""
        
        let titlePattern = "<title>(.*?)</title>"
        let regex = try NSRegularExpression(pattern: titlePattern)
        let range = NSRange(html.startIndex..., in: html)
        
        var name = "Imported Recipe"
        if let match = regex.firstMatch(in: html, range: range) {
            if let titleRange = Range(match.range(at: 1), in: html) {
                name = String(html[titleRange])
            }
        }
        
        return [
            "name": name,
            "ingredients": [],
            "instructions": [],
            "prepTime": 0,
            "cookTime": 0,
            "servings": 1
        ]
    }
}

File: android/app/src/main/java/com/neostudios/zest/data/remote/RecipeImportService.kt
package com.neostudios.zest.data.remote

import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.withContext
import java.net.URL
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class RecipeImportService @Inject constructor() {
    
    suspend fun importFromUrl(url: String): Map<String, Any> = withContext(Dispatchers.IO) {
        val html = URL(url).readText()
        val titleRegex = "<title>(.*?)</title>".toRegex()
        val name = titleRegex.find(html)?.groupValues?.get(1) ?: "Imported Recipe"
        
        mapOf(
            "name" to name,
            "ingredients" to emptyList<String>(),
            "instructions" to emptyList<String>(),
            "prepTime" to 0,
            "cookTime" to 0,
            "servings" to 1
        )
    }
}
